projects/plugins/boost/app/lib/critical-css/source-providers/providers/Singular_Post_Provider.php

```diff
@@ -105,12 +105,23 @@ public static function describe_key( $provider_key ) { // phpcs:ignore Generic.C
 	 * @return mixed|void
 	 */
 	public static function get_post_types() {
-		$post_types = get_post_types( array( 'public' => true ) );
+		$post_types = get_post_types(
+			array(
+				'public' => true,
+			),
+			'objects'
+		);
 		unset( $post_types['attachment'] );
 
 		$post_types = array_filter( $post_types, 'is_post_type_viewable' );
 
-		return apply_filters( 'jetpack_boost_critical_css_post_types', $post_types );
+		$provider_post_types = array();
+		// Generate a name => name array for backwards compatibility.
+		foreach ( $post_types as $post_type ) {
+			$provider_post_types[ $post_type->name ] = $post_type->name;
+		}
+
+		return apply_filters( 'jetpack_boost_critical_css_post_types', $provider_post_types, $post_types );
```

Comment:
maybe it would be better to restructure the code so that the filter `jetpack_boost_critical_css_post_types` is only applied in one place (method) - since the filter is going to be deprecated - it's going to make the deprecation experience smoother I think.

Reply:
Well that's the problem. Previously it was used in two places and I think incorrectly. I introduced `jetpack_boost_critical_css_post_types_archives` for the archives provider, but the singular post provider will still use `jetpack_boost_critical_css_post_types`.

So technically, the incorrect usage of `jetpack_boost_critical_css_post_types` in the archive provider is being fixed. I'm not sure how to handle a situation like this.

Reply:
Updated in 3198ec8.